packages/invoices/src/OutboundShipment/DetailView/modals/hooks/usePackSizeController.tsx

```diff
@@ -0,0 +1,87 @@
+import { useState, useEffect, useMemo } from 'react';
+import {
+  useTranslation,
+  ifTheSameElseDefault,
+} from '@openmsupply-client/common';
+import { DraftOutboundLine } from '../../../../types';
+
+export type PackSizeController = ReturnType<typeof usePackSizeController>;
+
+const distinctSortedPackSizes = (lines: DraftOutboundLine[]): number[] =>
+  Array.from(
+    new Set(
+      lines
+        .filter(
+          ({ onHold, availableNumberOfPacks }) =>
+            availableNumberOfPacks > 0 && !onHold
+        )
+        .reduce((sizes, { packSize }) => [...sizes, packSize], [] as number[])
+        .sort((a: number, b: number) => a - b)
+    )
+  );
+
+const usePackSizeOptions = (
+  packSizes: number[]
+): { label: string; value: number }[] => {
+  const t = useTranslation('distribution');
+  const anySize: { label: string; value: number }[] = [];
+  if (packSizes.length > 1) {
+    anySize.push({ label: t('label.any'), value: -1 });
+  }
+
+  const options = useMemo(
+    () =>
+      anySize.concat(
+        packSizes.map(packSize => ({
+          label: String(packSize),
+          value: packSize,
+        }))
+      ),
+    [packSizes, anySize]
+  );
+
+  return options;
+};
+
+export const usePackSizeController = (lines: DraftOutboundLine[]) => {
+  const packSizes = useMemo(() => distinctSortedPackSizes(lines), [lines]);
+  const options = usePackSizeOptions(packSizes);
+
+  const [selected, setSelected] = useState({ label: '', value: 0 });
+
+  const setPackSize = (newValue: number) => {
+    const packSizeOption = options.find(({ value }) => value === newValue);
+    if (!packSizeOption) return;
+    setSelected(packSizeOption);
+  };
+
+  useEffect(() => {
+    // if (selected.value !== 0) return;
```

Comment:
with this commented out, the modal re-renders continually and I can't enter quantities for anything 😬 
I improved things in main, so it might not be a problem after unpicking those conflicts? I can have a look after merging these changes in if you like?